src/types/routeMethods.ts

```diff
@@ -22,21 +22,21 @@ type RouteMethodsTuple<
 
 type RouteMethodsOrMethod<
   TRoute extends Route,
-  TPathParams extends Record<string, unknown>,
-  TQueryParams extends Record<string, unknown>
-> = TRoute extends { children: infer Children }
-  ? Children extends Routes
-    ? TRoute extends Public<TRoute>
-      ? RouteMethods<Children, RoutePathParams<TRoute, TPathParams>, RouteQueryParams<TRoute, TQueryParams>> & CreateRouteMethod<MergeParams<RoutePathParams<TRoute, TPathParams>, RouteQueryParams<TRoute, TQueryParams>>>
-      : RouteMethods<Children, RoutePathParams<TRoute, TPathParams>, RouteQueryParams<TRoute, TQueryParams>>
-    : never
+  TPathParams extends Record<string, unknown[]>,
+  TQueryParams extends Record<string, unknown[]>
+> = TRoute extends { children: infer Children extends Routes }
+  ? TRoute extends Public<TRoute>
+    ? RouteMethods<Children, RoutePathParams<TRoute, TPathParams>, RouteQueryParams<TRoute, TQueryParams>> & CreateRouteMethod<TRoute, TPathParams, TQueryParams>
+    : RouteMethods<Children, RoutePathParams<TRoute, TPathParams>, RouteQueryParams<TRoute, TQueryParams>>
   : TRoute extends Public<TRoute>
-    ? CreateRouteMethod<MergeParams<RoutePathParams<TRoute, TPathParams>, RouteQueryParams<TRoute, TQueryParams>>>
+    ? CreateRouteMethod<TRoute, TPathParams, TQueryParams>
     : never
 
```

Comment:
Simplified this type a bit by removing a ternary and also letting `CreateRouteMethod` do the merging of params.